### DISCUSSION TITLE
[Question] How do you add the Selector Manager events/actions to the Undo Manager stack?

### DISCUSSION THREAD

**Discussion by mortimerz** [GrapesJS/grapesjs#3877] [Q&A]
Hello everyone, 

I'm new to grapesjs and I'm currently working with Undo Manager. I'm trying to find a way on how to be able to undo/redo the action when adding new class selectors, does anyone have an idea or a working sample on how to add selector manager events (specifically adding class selectors) on the Undo Manager stack?

If my understanding is correct, the undoManager.add() method accepts a Model/Entity and not an action,  so my assumption is I need to add the model of the SelectorManager to the UndoManager stack to be able to track it, but I'm not sure if that's the correct way. 

TIA

**Comment by artf** [comment]
Hi @mortimerz this is actually my bad for not tracking selectors by default. 
Your understanding of UndoManager is correct but I'd avoid trying to extend it right now, this is another module that I should revisit and make more extendable.
I'd suggest opening the bug issue in regard to selectors and I'll try to fix it for the next release.

**Reply by mortimerz** [reply]
Hello @artf , thank you for your response. Sure, I'll open a bug issue for this one.

**Reply by mortimerz** [reply]
Hello again @artf - I've opened a [bug issue](https://github.com/artf/grapesjs/issues/3879 ) as suggested. I was also wondering, do you already have something in mind on how to address the issue? How can we contribute back related to this?

**Reply by artf** [reply]
Thanks @mortimerz I'm not yet sure but I guess it should be enough registering selectors here: 
https://github.com/artf/grapesjs/blob/b7492efd3f2adfbd02df9b65dca0dae37ef824ff/src/dom_components/model/Component.js#L200-L203
Something like this:
```js
um.add(comps); // This one registers children components
um.add(this.get('classes'));
```
